src/components/layout/use-anchor-tag-scrolling.ts

```diff
@@ -7,18 +7,25 @@ export const useAnchorTagScrolling = (): void => {
       const target = document.querySelector(
         `a[href*='${window.location.hash}']`,
       );
-      if (target) {
-        const scrollTop =
-          target.getBoundingClientRect().top +
-          window.pageYOffset -
-          // we need to scroll past the header and a little offset
-          (Number.parseInt(HEADER_HEIGHT, 10) + 16);
-
-        window.scrollTo({
-          top: scrollTop,
-          behavior: 'smooth',
-        });
-      }
+      scrollToTarget(target);
     }
   }, []);
 };
+
+export const scrollToTarget = (target: Element | null) => {
```

Comment:
I see how you need access to that part of the functionality. Good to split and export it then.